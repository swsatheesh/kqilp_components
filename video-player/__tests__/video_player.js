import React from 'react';
import renderer from 'react-test-renderer';
import VideoPlayer from '../video_player';

import poster from '../media/poster.png';
import video from '../media/sample_video.mp4';

// Mock store setup.

jest.unmock('../video_player');

describe('components', () => {
  describe('<VideoPlayer />', () => {
    it('should render VideoPlayer', () => {
      const comp = renderer.create(
        <VideoPlayer
          src={video}
          poster={poster}
          autoPlay={false}
          onTimeUpdate={() => {
          }}
          controlBar
        />
      ).toJSON();
      expect(comp).toMatchSnapshot();
    });
  });
});
