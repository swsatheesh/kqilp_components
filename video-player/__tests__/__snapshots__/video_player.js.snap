// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <VideoPlayer /> should render VideoPlayer 1`] = `
<div
  className="video-react-controls-enabled video-react-paused video-react-fluid video-react-user-active video-react-workinghover video-react"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onTouchStart={[Function]}
  role="region"
  style={
    Object {
      "paddingTop": "56.25%",
    }
  }
  tabIndex="-1"
>
  <video
    autoPlay={false}
    className="video-react-video"
    crossOrigin={undefined}
    id={undefined}
    loop={false}
    muted={false}
    onAbort={[Function]}
    onCanPlay={[Function]}
    onCanPlayThrough={[Function]}
    onDurationChange={[Function]}
    onEmptied={[Function]}
    onEnded={[Function]}
    onError={[Function]}
    onLoadStart={[Function]}
    onLoadedData={[Function]}
    onLoadedMetadata={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onPlaying={[Function]}
    onProgress={[Function]}
    onRateChange={[Function]}
    onSeeked={[Function]}
    onSeeking={[Function]}
    onStalled={[Function]}
    onSuspend={[Function]}
    onTimeUpdate={[Function]}
    onVolumeChange={[Function]}
    onWaiting={[Function]}
    playsInline={true}
    poster="poster.png"
    preload="auto"
    src={undefined}
  />
  <div
    className="video-react-poster"
    onClick={[Function]}
    style={
      Object {
        "backgroundImage": "url(\\"poster.png\\")",
      }
    }
  />
  <div
    className="video-react-loading-spinner"
  />
  <button
    aria-live="polite"
    className="video-react-big-play-button video-react-big-play-button-center big-play-button-hide"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="video-react-control-text"
    >
      Play Video
    </span>
  </button>
  <div
    className="video-react-control-bar"
  >
    <button
      className="video-react-play-control video-react-control video-react-button video-react-paused"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="video-react-control-text"
      >
        Play
      </span>
    </button>
    <div
      className="video-react-current-time video-react-time-control video-react-control"
    >
      <div
        aria-live="off"
        className="video-react-current-time-display"
      >
        <span
          className="video-react-control-text"
        >
          Current Time 
        </span>
        0:00
      </div>
    </div>
    <div
      className="video-react-time-control video-react-time-divider"
      dir="ltr"
    >
      <div>
        <span>
          /
        </span>
      </div>
    </div>
    <div
      className="video-react-duration video-react-time-control video-react-control"
    >
      <div
        aria-live="off"
        className="video-react-duration-display"
      >
        <span
          className="video-react-control-text"
        >
          Duration Time 
        </span>
        0:00
      </div>
    </div>
    <div
      className="video-space"
    />
    <div
      className="video-react-progress-control video-react-control"
      onMouseMove={[Function]}
    >
      <div
        aria-label="video progress bar"
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow="NaN"
        aria-valuetext="0:00"
        className="video-react-progress-holder video-react-slider-horizontal video-react-slider"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
      >
        <div
          className="video-react-play-progress video-react-slider-bar"
          data-current-time="0:00"
          style={
            Object {
              "width": "NaN%",
            }
          }
        >
          <span
            className="video-react-control-text"
          >
            <span>
              Progress
            </span>
            : 
            NaN%
          </span>
        </div>
      </div>
    </div>
    <div
      className="video-react-volume-menu-button-vertical video-react-vol-3 video-react-volume-menu-button video-react-menu-button-popup video-react-control video-react-button video-react-menu-button"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="video-react-menu"
        onClick={[Function]}
      >
        <div
          className="video-react-menu-content"
        >
          <div
            aria-label="volume level"
            aria-valuemax={100}
            aria-valuemin={0}
            aria-valuenow="100.00"
            aria-valuetext="100.00%"
            className="video-react-volume-bar video-react-slider-bar video-react-slider-vertical video-react-slider"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
          >
            <div
              className="video-react-volume-level"
              style={
                Object {
                  "height": "100.00%",
                }
              }
            >
              <span
                className="video-react-control-text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      className="video-react-icon-fullscreen video-react-fullscreen-control video-react-control video-react-button video-react-icon"
      onClick={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="video-react-control-text"
      >
        Non-Fullscreen
      </span>
    </button>
  </div>
</div>
`;
